shared/fs/filepreview/index.js

```diff
@@ -1,131 +1,51 @@
-        <Icon type="iconfont-ellipsis" style={stylesHeaderIcon} onClick={props.onAction} />
-      </Box>
-    </Box>
-    <Box style={stylesGreyContainer}>
-      <Box style={stylesContainer}>
-        <PathItemIcon spec={props.itemStyles.iconSpec} style={{}} />
-        <Text type="BodyBig" style={stylesFilename(props.itemStyles.textColor)}>
-          {props.pathItem.name}
-        </Text>
-        <Text type="BodySmall">{Constants.humanReadableFileSize(props.pathItem.size)}</Text>
-        {isMobile && (
-          <PathItemInfo
-            lastModifiedTimestamp={props.pathItem.lastModifiedTimestamp}
-            lastWriter={props.pathItem.lastWriter.username}
-            startWithLastModified={true}
-          />
-        )}
-        {isMobile && (
-          <Text type="BodySmall" style={stylesNoOpenMobile}>
-            This document can not be opened on mobile. You can still interact with it using the ••• menu.
-          </Text>
-        )}
-        {// Enable this button for desktop when we have in-app sharing.
-        isMobile && (
-          <Button
-            key="share"
-            type="Primary"
-            label="Share"
-            style={{marginTop: globalMargins.medium}}
-            onClick={props.onShare}
-          />
-        )}
-        {isIOS ? (
-          Constants.isMedia(props.pathItem.name) && (
-            <Button
-              key="open"
-              type="Secondary"
-              label={'Save'}
-              style={{marginTop: globalMargins.small}}
-              onClick={props.onSave}
-            />
-          )
-        ) : props.fileUIEnabled ? (
-          <Button
-            key="open"
-            type="Secondary"
-            label={'Show in ' + fileUIName}
-            style={{marginTop: globalMargins.small}}
-            onClick={props.onShowInFileUI}
-          />
-        ) : (
-          <Button
-            key="open"
-            type="Secondary"
-            label="Download a copy"
-            style={{marginTop: globalMargins.small}}
-            onClick={props.onDownload}
-          />
-        )}
-      </Box>
-    </Box>
-    <Footer />
-  </Box>
-)
-
-const stylesClose = {
-  marginLeft: globalMargins.tiny,
+const getDisplayComponent = (path: Types.Path, fileViewType: Types.FileViewType) => {
+  switch (fileViewType) {
+    case 'text':
+      return <TextView url="https://keybase.io/warp/release.txt" />
+    case 'default':
+      return <DefaultView path={path} />
+    default:
+      // eslint-disable-next-line no-unused-expressions
+      ;(fileViewType: empty) // this breaks when a new file view type is added but not handled here
+      return <Text type="BodyError">This shouldn't happen</Text>
+  }
 }
 
-const stylesCommonCore = {
-  alignItems: 'center',
-  borderBottomColor: globalColors.black_05,
-  borderBottomWidth: 1,
-  justifyContent: 'center',
+const FilePreview = ({routeProps}: FilePreviewProps) => {
```

Comment:
i don't think we should use routeprops or a generic 'bag' going down to component

Reply:
Do you mean we should only have one thing in `routeProps` or just in general we should only use `routeProps` in containers?

I think for this instance, `fileViewType` is actually not in `routeProps`. I'll fix the line below.

Reply:
routeProps should only be known to the routed component and shouldn't pass that down. we want to isolate what knows about this structure (especially as we're likely going to change it)

Reply:
views should take view specific props that are simple and about the view itself and not leak these kinds of abstractions (aka views shouldn't know about immutable, etc)

Reply:
Oh I see; in this case this component is indeed what's being referenced in `fs/routes.js`.

Reply:
Just realized you had a second comment after that. The base types for the two fields are string and boolean in this case. You mean we should just use string and boolean rather than types from `constants/type/fs`?

(I'm realizing fileViewType is actually needed in this case. I guess I'll fix it in the next PR to avoid conflicts.)